editions/tw5.com/tiddlers/releasenotes/5.4.0/#9084.tid

```diff
@@ -0,0 +1,27 @@
+title: $:/changenotes/5.4.0/#9084
+description: Optimize Freelinks plugin with Aho-Corasick for large wikis and non-Latin languages
+tags: $:/tags/ChangeNote
+change-type: improvement
+change-category: core
+github-links: [https://github.com/TiddlyWiki/TiddlyWiki5/pull/9084](https://github.com/TiddlyWiki/TiddlyWiki5/pull/9084)
+github-contributors: s793016
+
+This change introduces a fully optimized override of the core text widget, integrating an enhanced Aho-Corasick algorithm for automatic linkification of tiddler titles within text (freelinks). The new implementation prioritizes performance for large wikis and correct support for non-Latin scripts such as Chinese.
+
+**Highlights:**
```

Comment:
Don't add **
When generating things, make sure to check the result manually to remove those AI signatures.